config/src/config/network_config.rs

```diff
@@ -121,14 +108,22 @@ impl NetworkConfig {
             self.listen_address = utils::get_local_ip().ok_or_else(|| anyhow!("No local IP"))?;
         }
 
-        let keypair = NetworkKeyPairs::default();
-        let default_peer_id = PeerId::try_from(keypair.identity_keys.public().to_bytes()).unwrap();
-        let identity_key = self.network_keypairs.identity_keys.public();
-        let key_peer_id = PeerId::try_from(identity_key.to_bytes()).unwrap();
+        if self.enable_remote_authentication {
+            ensure!(
+                self.enable_noise,
+                "For a node to enforce remote authentication, noise must be enabled.",
+            );
+        }
 
-        // If PeerId is not set, it is derived from NetworkIdentityKey.
-        if self.peer_id == default_peer_id {
-            self.peer_id = key_peer_id;
+        // If PeerId is not set but NetworkIdentityKey exists, derive the PeerId.
+        if self.peer_id == PeerId::default() && self.network_keypairs.is_some() {
+            let identity_key = self
```

Comment:
make a helper function as you're doing this many times.

Reply:
Thanks! I've actually refactored these checks to avoid the duplication and better compact the logic around PeerId and IdentityKey. Personally, I think it's clearer than the previous logic, and makes the checks more explicit.